### PR TITLE
Add Org-Mode lexer

### PR DESCRIPTION
**Pull request by goshatch**
Hello,

This PR adds a new lexer for the Emacs [Org-Mode](https://orgmode.org). It is basically a direct translation of [the work](https://github.com/alecthomas/chroma/pull/156) done by @kaushalmodi for Go's Chroma library, which I came across in https://github.com/pygments/pygments/issues/426#issuecomment-526846873.

I have to say that I am familiar with neither Python nor Go; my reason for wanting this lexer is so that my forge of choice (Sourcehut) which uses `pygments` would syntax highlight my org files 😁 

~~Therefore, I would like to ask for help on this PR, and I am marking it as draft. Currently, the test for the lexer is failing (I'm adding the output below). I don't know enough about the Python ecosystem to debug it myself, and I hope someone here would be able to assist.~~

Thank you for your consideration!

### CODE REVIEW

File: tests/examplefiles/org/example.org
Comment (goshatch) [inline tests/examplefiles/org/example.org:19]: The lexer trips up here: the test output does not contain the blank line (l18). It's possible that it's confusing `*bold*` with a heading line (`* heading`), but I am unsure how to fix this.

Comment (goshatch) [reply tests/examplefiles/org/example.org:19]: This is now fixed!

File: pygments/lexers/markup.py
Comment (goshatch) [inline pygments/lexers/markup.py:669]: This could certainly be improved to use the appropriate lexer to highlight the code inside the code block (like it's done in the Markdown lexer), but this is a bit beyond my expertise. I would welcome contributions.

The usual code block format is like this (lifted from [my literal Emacs config](https://git.sr.ht/~gosha/dotfiles/tree/main/item/.doom.d/config.org#L11)):

```org-mode
#+begin_src emacs-lisp :tangle yes
(setq undo-limit 80000000
      evil-want-fine-undo t
      truncate-string-ellipsis "…")
#+end_src
```

**Review by jeanas — CHANGES_REQUESTED** [review]
(I only had the time to start a review.)

File: pygments/lexers/markup.py
Comment (goshatch) [reply pygments/lexers/markup.py:631]: Thank you for the review! 

I realise this MIME type is not listed at the link above, however it does seem to be the consensus for org-mode files, as per this mailing list message: https://lists.gnu.org/r/emacs-orgmode/2017-09/msg00087.html (the author of this message is the maintainer of Org Mode)

**Comment by goshatch** [comment]
> (I only had the time to start a review.)

Thank you @jeanas, I appreciate your time! I've made the changes you've requested, please let me know if anything else could be improved.

File: pygments/lexers/markup.py
Comment (jeanas) [reply pygments/lexers/markup.py:631]: Ah, I see that you are right:

```
$ echo "foobar" > foo.org

$ xdg-mime query filetype foo.org
text/org
```

**Comment by jeanas** [comment]
@goshatch There are a few things I'd like to fix up in the PR before merging it, but I cannot push code here because you opened the PR from the `master` branch of your fork, which is write-protected. I'm going to create a different PR. (For the future, it is normally better on GitHub to create a feature branch for each PR.)

**Comment by goshatch** [comment]
Thank you @jeanas, and noted about feature branches.

**Pull request closed without merging**
